docs/PATCH-ENVELOPES.md

```diff
@@ -96,3 +96,238 @@ All communication in this process is done via PubSub. When AppInstance downloads
+  ArtifactMetaData string `json:"artifactMetaData"`
+  // Encrypted ArtifactMetadata for blob
+  EncArtifactMeta CipherBlockStatus `json:"encArtifactMeta"`
+  // EncURL is URL to download encrypted binary blob in CipherBlockStatus format
+  // which contains either ONEOF inline or volume encrypted data
+  EncURL string `json:"encURL"`
+  // EncFileName is file name of the encrypted binary blob
+  EncFileName string `json:"encFileName"`
+  // Inline - used for post decrypt inline binary blob
+  Inline *BinaryBlobCompleted `json:"inline"`
+  // Volume - used for post decrypt volume binary blob
+  Volume *BinaryBlobVolumeRef `json:"volume"`
+}
+```
+
+* The EncType field indicates whether the encryption type is BlobEncryptedTypeInline or BlobEncryptedTypeVolume.
+* The ArtifactMetaData and EncArtifactMeta fields are similar to the existing BinaryBlobCompleted and BinaryBlobVolumeRef structures.
+* The EncURL field points to the cached directory containing the saved cipherData.
+* The EncFileName field specifies the key or filename for the saved cipherData.
+
+#### **Publishing the Patch Envelope**
+
+When the zedagent publishes the PatchEnvelopeInfo, the Inline and Volume fields are set to nil. The EncURL contains the cached URL for the encrypted inline data.
+
+#### **Decryption on the MSRV Side**
+
+* When the Metadata Server (MSRV) receives the PatchEnvelopeInfo, it iterates through all binary blobs to check for EncArtifactMeta fields that require decryption. If decryption is required, the decrypted data is populated into the ArtifactMetaData field.
+* For BinaryCipherBlob data, the MSRV fetches the saved cipherData from the cache directory. Based on the type (inline or volumeref), it decrypts the data and populates the Inline or Volume structure.
+
+#### **Serving the Data**
+
+* For inline blobs, the MSRV dynamically formats the url string when a metadata client queries the description.json. This is handled by the zedagent parser for unencrypted cases.
+* When a client requests to download an inline file, the MSRV dynamically decrypts the data from the cache directory, retrieves the content, and forwards it to the client in the download response.
+
+#### **Serving Zip File Download**
+
+When a client requests a zip file containing the patch envelope, the GetZipArchive() utility function:
+
+* Decrypts the CipherBlobs.
+* Appends the decrypted data to the zip entries.
+* Includes the decrypted data in the final zip file for download.
+
+### Example of patch envelope from zedagent 'PatchEnvelopeInfoList' and client description.json
+
+Below is an example of a patch envelope containing one unencrypted BinaryBlob, one unencrypted VolumeRef, and three encrypted CipherBlobs with inline blobs. This example demonstrates how encrypted and unencrypted blobs coexist within a patch envelope.
+
+b6806473-ab5f-4807-828d-725dcddc45ec:/persist/status/zedagent/PatchEnvelopeInfoList# jq < global.json
```

Comment:
This line can also be inside a shell block for better readability....

Reply:
Done.